build/common/installer/scripts/fluent-bit-geneva-conf-customizer.rb

```diff
@@ -0,0 +1,83 @@
+#!/usr/local/bin/ruby
+require_relative "ConfigParseErrorLogger"
+
+@fluent_bit_conf_path = "/etc/opt/microsoft/docker-cimprov/fluent-bit-geneva.conf"
+
+@os_type = ENV["OS_TYPE"]
+if !@os_type.nil? && !@os_type.empty? && @os_type.strip.casecmp("windows") == 0
+  @fluent_bit_conf_path = "/etc/fluent-bit/fluent-bit-geneva.conf"
+end
+
+@default_service_interval = "15"
+@default_mem_buf_limit = "10"
+
+def is_number?(value)
+  true if Integer(value) rescue false
+end
+
+def substituteFluentBitPlaceHolders
```

Comment:
why we sperate the scripts into 3 instead of putting into 1 with different functions?

Reply:
some of the checks can be reused  like line 48 - 63

Reply:
Good feedback. Fixed it.

Reply:
Don't see this fixed? We still have 3 files doing the same thing?